### PR TITLE
Pyupgrade 3.6: Update syntax with Python 3.6+ features

### PR DESCRIPTION
**Pull request by EwoutH**
Updated Python code with Python 3.6+ features:
- utf-8 encoding is now the default ([PEP 3120](https://www.python.org/dev/peps/pep-3120/), Python 3.0+)
- Replace list comprehensions by Generator Expressions ([PEP 289](https://www.python.org/dev/peps/pep-0289/), Python 2.4+)
- Replace `yield` loop by `yield from` ([PEP 380](https://www.python.org/dev/peps/pep-0380/), Python 3.3+)
- Replace the IOError alias by OSError ([PEP 3151](https://www.python.org/dev/peps/pep-3151/), Python 3.3+)
- Remove the default subclass `(object)` when defining a class
- Define sets with curly braces `{}` instead of `set()`
- Remove `"r"` parameter from open function, which is default

Used [pyupgrade](https://github.com/asottile/pyupgrade) 2.29.1 with `--py36-plus` as starting point for this PR.

### CODE REVIEW

File: selfdrive/controls/radard.py [hunk 1]
```diff
@@ -135,7 +135,7 @@ def update(self, sm, rr, enable_lead):
       self.tracks[ids].update(rpt[0], rpt[1], rpt[2], v_lead, rpt[3])
 
     idens = list(sorted(self.tracks.keys()))
-    track_pts = list([self.tracks[iden].get_key_for_cluster() for iden in idens])
+    track_pts = list(self.tracks[iden].get_key_for_cluster() for iden in idens)
```
Comment (GregorKikelj) [inline selfdrive/controls/radard.py:139]: Could list be replaced with [] here?

File: selfdrive/controls/radard.py [hunk 2]
```diff
@@ -135,7 +135,7 @@ def update(self, sm, rr, enable_lead):
...
```
Comment (GregorKikelj) [inline selfdrive/controls/radard.py:138]:
```suggestion
    track_pts = [self.tracks[iden].get_key_for_cluster() for iden in idens]
```

Comment (EwoutH) [reply selfdrive/controls/radard.py]: Good catch, implemented it.

**Comment by EwoutH** [comment]
Thanks for the suggestions, I implemented them, squashed the commits, rebased and signed it.

File: selfdrive/test/process_replay/compare_logs.py [hunk 3]
```diff
@@ -58,7 +58,7 @@ def compare_logs(log1, log2, ignore_fields=None, ignore_msgs=None, tolerance=Non
   if ignore_msgs is None:
     ignore_msgs = []
 
-  log1, log2 = [list(filter(lambda m: m.which() not in ignore_msgs, log)) for log in (log1, log2)]
+  log1, log2 = (list(filter(lambda m: m.which() not in ignore_msgs, log)) for log in (log1, log2))
```
Comment (GregorKikelj) [inline selfdrive/test/process_replay/compare_logs.py:61]: I think this should also work.
```suggestion
  log1, log2 = ([filter(lambda m: m.which() not in ignore_msgs, log)] for log in (log1, log2))
```

Comment (GregorKikelj) [reply selfdrive/test/process_replay/compare_logs.py:61]: Btw squashing is done automatically on merge, don't think it's needed in a PR

**Comment by adeebshihadeh** [comment]
Can you fix the failing tests?

File: selfdrive/test/process_replay/compare_logs.py (hunk 3 above)
Comment (GregorKikelj) [reply selfdrive/test/process_replay/compare_logs.py]: Sorry, though that [] and list() were interchangeable, apparently not in this case

**Comment by EwoutH** [comment]
I think replacing the `list()` by `[]` as @GregorKikelj suggested didn't play well with the lambda, so I reverted it. I did this a bit to quick, and accidentally reset the whole branch. Luckily `git reflow` is a thing and I recovered it. I couldn't get this PR reopened though, so it's now #23305.

(note to self: be really careful touching `git reset --hard`)

**Pull request closed without merging**